components/tikv_util/src/security.rs

```diff
@@ -166,6 +173,64 @@ impl ServerCredentialsFetcher for Fetcher {
     }
 }
 
+/// Check peer CN with cert-allowed-cn field.
+/// Return true when the match is successful (support wildcard pattern).
+/// Skip the check when cert-allowed-cn is not set or the secure channel is not used.
+pub fn check_common_name(cert_allowed_cn: &HashSet<String>, ctx: &RpcContext) -> bool {
```

Comment:
Could you add an integration test to make sure it's actually working?

Reply:
Done